Update CPython version and SIP build configuration

HellAholic opened the pull request:
Bumps CPython dependency from 3.12.2 to 3.12.7 and modifies SIP build tool configuration to use the full path to sip-build from the CPython Scripts directory for improved reliability.
Requires:
https://github.com/Ultimaker/conan-ultimaker-index/pull/25

CURA-12814

HellAholic committed:
Update CPython version and SIP build configuration

Bumps CPython dependency from 3.12.2 to 3.12.7 and modifies SIP build tool configuration to use the full path to sip-build from the CPython Scripts directory for improved reliability.

wawanbreton commented on conanfile.py:
```diff
@@ -120,7 +120,9 @@ def generate(self):
-        sip.configure()
+        # Use the full path to sip-build from the CPython Scripts directory
+        sip_build_path = os.path.join(self.dependencies["cpython"].cpp_info.bindirs[0], "Scripts", "sip-build.exe")
+        sip.configure(sip_install_executable=sip_build_path)
```
I'm afraid this is not going to work on non-windows OSes 😄 
Maybe there should be a change in `sipbuildtool` instead ?

HellAholic committed:
Improve sip-build path detection for cross-platform use

Replaces hardcoded Windows-specific sip-build path with auto-detection using the CPython dependency, making the SIP build process cross-platform compatible.

HellAholic committed:
Bump conan package version to 5.12.0-alpha.0

Update conandata.yml version from 5.11.0 to 5.12.0-alpha.0 to mark a pre-release alpha. Requirements entry for arcus remains at 5.11.0.

HellAholic committed:
Set version according to semver

Patch version update in dependency -> patch version increase in package

HellAholic committed:
Merge branch 'main' into CURA-12814

rburema reviewed (approved):
LGTM
